src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -233,6 +242,11 @@ - (instancetype)initWithMTROTAImageTransferHandler:(MTROTAImageTransferHandler *
 {
     assertChipStackLockedByCurrentThread();
 
+    // For thread devices, we need to throttle sending the response to BlockQuery, if the query is processed, before kBdxThrottleIntervalInMsecs
+    // has elapsed to prevent the BDX messages spamming up the network. Get the timestamp at which we start processing the BlockQuery message.
+
+    __block uint64_t startBlockQueryHandlingTimestamp = chip::System::SystemClock().GetMonotonicMilliseconds64().count();
```

Comment:
`count()` should be a last resort; you are no losing typesafety.  I would suggest something like this:
```suggestion
    auto startBlockQueryHandlingTimestamp = System::SystemClock().GetMonotonicTimestamp()
```

Note that I don't think this needs to be `__block`, because you are not mutating it in the blocks, right?